### PR TITLE
feat(TimePicker-compat): allow custom children through render function

### PR DESCRIPTION
**Pull request by YuanboXue-Amber**
Allow custom children through render function and add a story.

Usage:
```tsx
<TimePicker>
  {({ options }) => {
    // custom options' rendering here
  }}
</TimePicker>;
```

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/d4664628ced0837e16fd0cb20c79ba66030ed33d'>d4664628ced0837e16fd0cb20c79ba66030ed33d</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-timepicker-compat-preview/src/components/TimePicker/TimePicker.types.ts [hunk 1]
```diff
@@ -116,6 +114,18 @@ export type TimePickerProps = Omit<
     | 'freeform'
   > &
   TimeFormatOptions & {
+    /**
+     * TimePicker has default children. It can be customized as either:
+     * 1. A single element or an array of elements
+     * 2. A render function that will receive properties and must return valid element(s) or null
+     * 3. null or undefined
+     */
+    children?:
```
Comment (ling1726) [inline packages/react-components/react-timepicker-compat-preview/src/components/TimePicker/TimePicker.types.ts:123]: is it a requirement to have custom children in the TimePicker?

Comment (YuanboXue-Amber) [reply packages/react-components/react-timepicker-compat-preview/src/components/TimePicker/TimePicker.types.ts:123]: Nope, but I think it's nice to have. Do you think we should hold off until someone request a feature?

Comment (ling1726) [reply packages/react-components/react-timepicker-compat-preview/src/components/TimePicker/TimePicker.types.ts:123]: Yes, it wasn't a feature in v8, so let's wait until it's requested

**Pull request closed without merging**